experiments/AMIP/modular/components/slab_init.jl

```diff
@@ -1,146 +0,0 @@
-# slab_rhs! (TODO: rm during clean up)
-using ClimaCore
```

Comment:
Not entirely familiar with all the files in the previous iterations: What was the functionality within this, and where exactly in the new modifications is it retained?

Reply:
`slab_init.jl` used to contain a general slab model initialization interface. Some time ago, we decided to use separate slab ocean and slab sea ice models, because their formulation diverged. So this slab model template became redundant. In this PR, the `slab_utils.jl` was revamped to contain only the currently used common functions for the two models, including the `dss!`.